src/application_settings/container_base.py

```diff
@@ -112,74 +104,149 @@ def filepath(cls) -> PathOpt:
-        return _the_config
+    else:
 
-    @classmethod
-    def update(
-        cls: type[_ContainerT], changes: dict[str, dict[str, Any]]
-    ) -> _ContainerT:
-        "Update and save the settings with data specified in changes; not meant for config"
-        return cls.get()._update(changes)  # pylint: disable=protected-access
-
-    def _update(self: _ContainerT, changes: dict[str, dict[str, Any]]) -> _ContainerT:
-        "Update and save the settings with data specified in changes; not meant for config"
-        # filter out fields that are both in changes and an attribute of the SettingsContainer
-        _sections_to_update = {
-            fld for fld in fields(self) if fld.init and fld.name in changes.keys()
-        }
-
-        # update the sections and keep them in a dict
-        # actually sections: dict[str, _ContainerSectionT]
-        # but MyPy doesn't swallow that
-        updated_sections: dict[str, Any] = {
-            fld.name: _update_section(getattr(self, fld.name), changes[fld.name])
-            for fld in _sections_to_update
-        }
-        new_settings = replace(self, **updated_sections)
-        new_settings._set()  # pylint: disable=protected-access
-        new_settings._save()  # pylint: disable=protected-access
-        return new_settings
+        @classmethod
+        def get(cls: type[Self], reload: bool = False) -> Self:
+            """Get the singleton; if not existing, create it."""
 
-    @classmethod
-    def _create_instance(cls: type[_ContainerT]) -> _ContainerT:
-        """Instantiate the Container."""
-
-        # get whatever is stored in the config/settings file
-        data_stored = cls._get_stored_data()
-        # filter out fields that are both stored and an attribute of the Container
-        _data_fields = {
-            fld for fld in fields(cls) if fld.init and fld.name in data_stored.keys()
-        }
-        # instantiate the sections and keep them in a dict
-        # actually sections: dict[str, _ContainerSectionT]
-        # but MyPy doesn't swallow that
-        sections: dict[str, Any] = {
-            fld.name: cls._instantiate_section(fld.type, data_stored[fld.name])
-            for fld in _data_fields
-        }
-
-        # instantiate the Container with the sections
-        return cls(**sections)
+            if (_the_container_or_none := cls._get()) is None or reload:
+                # no config has been made yet or it needs to be reloaded,
+                # so let's instantiate one and keep it in the global store
+                return cls._create_instance()
+            return _the_container_or_none
+
+        @classmethod
+        def update(cls: type[Self], changes: dict[str, dict[str, Any]]) -> Self:
+            "Update and save the settings with data specified in changes; not meant for config"
+            return cls.get()._update(changes)  # pylint: disable=protected-access
+
+        if sys.version_info < (3, 10):
```

Comment:
we even need another level?

Reply:
Tried to improve by having the 3.9 special case right next to the Self TypeVar def. Because of the general meaning of Self and the bound in this TypeVar def, we cannot move this to type_notation_helper.py